### ISSUE TITLE
Multi Row Insert fails with "Failed to decrypt. Check file key and master key."

### ISSUE THREAD

**Issue by markddrake** [snowflakedb/snowflake-connector-nodejs#374]
```
C:\Development\bind>node --version
v18.9.0

C:\Development\bind>npm --version
7.21.0

C:\Development\bind>systeminfo | findstr /B /C:"OS Name" /C:"OS Version"
OS Name:                   Microsoft Windows 11 Pro
OS Version:                10.0.22621 N/A Build 22621
```
The following code 
```
import snowflake from 'snowflake-sdk';

import fs                 from "fs"

import { 
  performance 
}                         from 'perf_hooks';

const ddl = `create table if not exists "DEMO_DB"."YTEST"."CUSTOMERS"(
  "CUST_ID" NUMBER(38,19) ,"CUST_FIRST_NAME" TEXT ,"CUST_LAST_NAME" TEXT ,"CUST_GENDER" TEXT ,"CUST_YEAR_OF_BIRTH" NUMBER(4) ,"CUST_MARITAL_STATUS" TEXT ,"CUST_STREET_ADDRESS" TEXT ,"CUST_POSTAL_CODE" TEXT ,"CUST_CITY" TEXT ,"CUST_CITY_ID" NUMBER(38,19) ,"CUST_STATE_PROVINCE" TEXT ,"CUST_STATE_PROVINCE_ID" NUMBER(38,19) ,"COUNTRY_ID" NUMBER(38,19) ,"CUST_MAIN_PHONE_NUMBER" TEXT ,"CUST_INCOME_LEVEL" TEXT ,"CUST_CREDIT_LIMIT" NUMBER(38,19) ,"CUST_EMAIL" TEXT ,"CUST_TOTAL" TEXT ,"CUST_TOTAL_ID" NUMBER(38,19) ,"CUST_SRC_ID" NUMBER(38,19) ,"CUST_EFF_FROM" TIMESTAMP_NTZ(9) ,"CUST_EFF_TO" TIMESTAMP_NTZ(9) ,"CUST_VALID" TEXT )`
  
const dml = `insert into "DEMO_DB"."YTEST"."CUSTOMERS" ("CUST_ID","CUST_FIRST_NAME","CUST_LAST_NAME","CUST_GENDER","CUST_YEAR_OF_BIRTH","CUST_MARITAL_STATUS","CUST_STREET_ADDRESS","CUST_POSTAL_CODE","CUST_CITY","CUST_CITY_ID","CUST_STATE_PROVINCE","CUST_STATE_PROVINCE_ID","COUNTRY_ID","CUST_MAIN_PHONE_NUMBER","CUST_INCOME_LEVEL","CUST_CREDIT_LIMIT","CUST_EMAIL","CUST_TOTAL","CUST_TOTAL_ID","CUST_SRC_ID","CUST_EFF_FROM","CUST_EFF_TO","CUST_VALID")  values   (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)`

const dataset = JSON.parse(fs.readFileSync('dataset.json','utf-8'));

class Tester {

   constructor() {
    
	  this.connectionProperties =  {
  	    "account"                       : "****"
      , "username"                      : "****"
      , "password"                      : "****"
	  , "warehouse"                     : "DEMO_WH"
      , "database"                      : "DEMO_DB"
	  , "schema"                        : "PUBLIC"	   
    }
  }
   
  establishConnection(connection) {
      
    return new Promise((resolve,reject) => {
      connection.connect((err,connection) => {
        if (err) {
          reject(err)
        }
        resolve(connection)
      })
    })
  } 
  
  executeSQL(sqlStatement,binds) {
    
    return new Promise((resolve,reject) => {

	  this.connection.execute({
        sqlText        : sqlStatement
      , binds          : binds
	  , fetchAsString  : ['Number','Date','JSON']
      , complete       : async (err,statement,rows) => {
                           if (err) reject(err)
                           resolve(rows)
	                     }   
      })
    })
  }  
  
  async runTest() {

	 try {
       this.connection = snowflake.createConnection(this.connectionProperties)
       this.connection = await this.establishConnection(this.connection)
       await this.executeSQL('DROP SCHEMA IF EXISTS YTEST');
       await this.executeSQL('CREATE SCHEMA YTEST');   	  
       await this.executeSQL(ddl);
	   console.log('DDL')
	   await this.executeSQL(dml,dataset);
	   console.log('DML')
       const results = await this.executeSQL(`select count(*) from "DEMO_DB"."YTEST"."CUSTOMERS"`);	   
	   console.log(results)
	 } catch (e) {
	   console.log(e);
	 }
   }

}

async function main() {
  const t = new Tester()
  await t.runTest()
}

main().then(() => {console.log('Success')}).catch((e) => {console.log(e)})
```

fails intemittently with 

````
C:\Development\bind>node test.js
DDL
DML
[ { 'COUNT(*)': '5200' } ]
Success

C:\Development\bind>node test.js
DDL
DML
[ { 'COUNT(*)': '5200' } ]
Success

C:\Development\bind>node test.js
DDL
Error [OperationFailedError]: 
    at createError (C:\Development\bind\node_modules\snowflake-sdk\lib\errors.js:537:15)
    at exports.createOperationFailedError (C:\Development\bind\node_modules\snowflake-sdk\lib\errors.js:316:10)
    at Object.callback (C:\Development\bind\node_modules\snowflake-sdk\lib\services\sf.js:647:28)
    at C:\Development\bind\node_modules\snowflake-sdk\lib\http\base.js:111:25
    at done (C:\Development\bind\node_modules\urllib\lib\urllib.js:589:5)
    at C:\Development\bind\node_modules\urllib\lib\urllib.js:953:9
    at decodeContent (C:\Development\bind\node_modules\urllib\lib\urllib.js:740:14)
    at handleResponseCloseAndEnd (C:\Development\bind\node_modules\urllib\lib\urllib.js:924:7)
    at IncomingMessage.<anonymous> (C:\Development\bind\node_modules\urllib\lib\urllib.js:962:7)
    at IncomingMessage.emit (node:events:525:35) {
  code: '100109',
  sqlState: 'XX000',
  data: {
    internalError: false,
    errorCode: '100109',
    age: 0,
    sqlState: 'XX000',
    queryId: '01a8bc54-0402-3ffc-000d-54830134349a'
  }
}
Success
```

The problem has been observed using 1.6.16.  It has not been observed using 1.6.13

**Comment by markddrake** [comment]
[dataset.zip](https://github.com/snowflakedb/snowflake-connector-nodejs/files/10147885/dataset.zip)

**Comment by sfc-gh-dszmolka** [comment]
while the debug effort is ongoing, the current latest version of snowflake-sdk (1.6.17) now has a connection option of `arrayBindingThreshold` (default value is 100000 if this option is not set). When configured in the connection, this can be set to a higher value now than the default which might help avoiding or reducing the number of occassions hitting the error by allowing a bigger number of binds to be sent to the statement before the client switches to the '[create temporary stage + PUT files before ingesting](https://github.com/snowflakedb/snowflake-connector-nodejs/blob/v1.6.17/lib/connection/statement.js#L94-L108)' path, which path occasionally leads to the issue. 

While this helps avoiding the problem, is not considered a final solution which the driver team is still investigating. Thank you folks for bearing with us while this is fixed.

**Comment by sfc-gh-dszmolka** [comment]
#398

**Comment by sfc-gh-dszmolka** [comment]
fixed with #419 , which is released with `v.1.6.19` yesterday. according to internal tests, it should really fix the issue but I encourage everyone who can to please upgrade the driver and take a stab at it and see if it indeed fixes the issue.

if you still have the same issue, please reopen/comment. closing this for now and thanks to everyone for bearing with us while this was fixed.
